zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessorTest.java

```diff
@@ -216,4 +217,126 @@ record -> {
                       });
             });
   }
+
+  @Test
+  void shouldReconstructForSingleFormWithoutDeploymentKey() {
+    // given
+    final var command = mock(TypedRecord.class);
+    when(command.getValue()).thenReturn(new DeploymentRecord());
+
+    final var formKey = Protocol.encodePartitionId(1, 2);
+    state
+        .getFormState()
+        .storeFormInFormColumnFamily(
+            new FormRecord()
+                .setFormKey(formKey)
+                .setFormId("form")
+                .setResourceName("form.form")
+                .setVersion(1));
+
+    // when
+    processor.processRecord(command);
+
+    // then
+    assertThat(resultBuilder.getFollowupRecords())
+        .singleElement()
+        .satisfies(
+            record -> {
+              assertThat(record.getIntent()).isEqualTo(DeploymentIntent.RECONSTRUCTED);
+              assertThat(record.getKey()).isEqualTo(formKey);
+              assertThat(record.getValue())
+                  .asInstanceOf(InstanceOfAssertFactories.type(DeploymentRecord.class))
+                  .satisfies(
+                      deploymentRecord -> {
+                        assertThat(deploymentRecord.getFormMetadata()).hasSize(1);
+                        assertThat(deploymentRecord.getDeploymentKey()).isEqualTo(formKey);
+                      });
+            });
+  }
+
+  @Test
+  void shouldReconstructForSingleFormWithDeploymentKey() {
+    // given
+    final var command = mock(TypedRecord.class);
+    when(command.getValue()).thenReturn(new DeploymentRecord());
+
+    final var deploymentKey = Protocol.encodePartitionId(1, 1);
+    final var formKey = Protocol.encodePartitionId(1, 2);
+    state
+        .getFormState()
+        .storeFormInFormColumnFamily(
+            new FormRecord()
+                .setDeploymentKey(deploymentKey)
+                .setFormKey(formKey)
+                .setFormId("form")
+                .setResourceName("form.form")
+                .setVersion(1));
+
+    // when
+    processor.processRecord(command);
+
+    // then
+    assertThat(resultBuilder.getFollowupRecords())
+        .singleElement()
+        .satisfies(
+            record -> {
+              assertThat(record.getIntent()).isEqualTo(DeploymentIntent.RECONSTRUCTED);
+              assertThat(record.getKey()).isEqualTo(deploymentKey);
+              assertThat(record.getValue())
+                  .asInstanceOf(InstanceOfAssertFactories.type(DeploymentRecord.class))
+                  .satisfies(
+                      deploymentRecord -> {
+                        assertThat(deploymentRecord.getFormMetadata()).hasSize(1);
+                        assertThat(deploymentRecord.getDeploymentKey()).isEqualTo(deploymentKey);
+                      });
+            });
+  }
+
+  @Test
+  void shouldIncludeAllFormsOfDeployment() {
```

Comment:
❓ Will you make a test in the end that combines different resources? So basically one that verifies if we have a process, that it can also find the forms and DRGs within the same deployment?

Reply:
We should have a test for this at the end :+1: